client/profile-wizard/steps/business-details/flows/selective-bundle/index.js

```diff
@@ -153,6 +129,47 @@ class BusinessDetails extends Component {
 			} );
 	}
 
+	async persistProfileItems( additions = {} ) {
+		const { updateProfileItems } = this.props;
+
+		const {
+			other_platform: otherPlatform,
+			other_platform_name: otherPlatformName,
+			product_count: productCount,
+			revenue,
+			selling_venues: sellingVenues,
+		} = this.state.savedValues;
+
+		const updates = {
+			other_platform: otherPlatform,
+			other_platform_name:
+				otherPlatform === 'other' ? otherPlatformName : '',
+			product_count: productCount,
+			revenue,
+			selling_venues: sellingVenues,
+			...additions,
+		};
+
+		// Remove possible empty values like `revenue` and `other_platform`.
+		const finalUpdates = Object.entries( updates ).reduce(
+			( acc, [ key, val ] ) => {
+				if ( val !== '' ) {
+					return {
+						...acc,
+						[ key ]: val,
+					};
+				}
+
+				return acc;
+			},
+			{}
+		);
+
+		return updateProfileItems( finalUpdates ).catch( () => {
+			throw new Error();
```

Comment:
I think we'll need to handle this error, possibly by creating an error notice.  This was previously handled by the promises array, but since `persistProfileItems` is called outside of that now, we need to make sure this error is also handled properly.

Reply:
Thanks @joshuatf , added in https://github.com/woocommerce/woocommerce-admin/pull/7426/commits/3201eb50d1eb0794527fd3bdc1c6b78d0be93eb3.